realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -227,6 +228,54 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
     }
 
     @Test
+    public void removePriorIndexesOrRenamingPrimaryKey() {
+        // Create v0 of the Realm
+        RealmConfiguration originalConfig = configFactory.createConfigurationBuilder()
+                .schema(AllJavaTypes.class)
+                .build();
+        Realm.getInstance(originalConfig).close();
+
+        RealmMigration migration = new RealmMigration() {
+            @Override
+            public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
+                realm.getSchema()
+                        .rename("AllJavaTypes", "PrimaryKeyAsLong")
+                        // There are three columns coming prior to the PK field 'fieldLong', and
+                        // removing "fieldString", "fieldShort" & "fieldInt" should not rearrange the PK
+                        .removeField("fieldString")
+                        .removeField("fieldShort")
+                        .removeField("fieldInt")
+
+                        // removal of rest fields should not affect PK attribute
```

Comment:
I was looking into it. But then I would create two more not much useful test entities that I avoided it. I'll create them and build proper unit tests.